### PR TITLE
fix: level role config

### PR DESCRIPTION
**Pull request by hoangharry**
**What does this PR do?**

-   [x] Fix catch case error record not found
-   [x] Add more log to entity layer

### CODE REVIEW

File: pkg/entities/default_role.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ func (e *Entity) GetDefaultRoleByGuildID(guildID string) (*response.DefaultRoleR
 }
 
 func (e *Entity) CreateDefaultRoleConfig(GuildID, RoleID string) error {
+	err := e.checkRoleIDBeenConfig(GuildID, RoleID)
```
Comment (trkhoi) [inline pkg/entities/default_role.go:28]: why need this here ?

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/entities/default_role.go (hunk 1 above)
Comment (hoangharry) [reply pkg/entities/default_role.go:28]: i think default role need to be different from other roles, so add here to check roleID has been used in any role configs, to make sure the default role is unique

File: pkg/entities/default_role.go [hunk 2]
```diff
@@ -25,6 +25,10 @@ func (e *Entity) GetDefaultRoleByGuildID(guildID string) (*response.DefaultRoleR
...
+	if err != nil {
+		e.log.Error(err, "[entities.CreateDefaultRoleConfig] check roleID been configed failed")
```
Comment (trkhoi) [inline pkg/entities/default_role.go:30]: should we return anything in error

File: pkg/entities/default_role.go (hunk 1 above)
Comment (trkhoi) [reply pkg/entities/default_role.go:28]: i think for now we dont need this flow, will add in future if needed

File: pkg/job/update_user_roles.go [hunk 3]
```diff
@@ -85,7 +86,14 @@ func (job *updateUserRoles) updateLevelRoles(guildID string) error {
...
 		userLevelRole, err := job.entity.GetUserRoleByLevel(guildID, userXP.Level)
-		if err != nil {
+		switch {
+		case err == gorm.ErrRecordNotFound:
+			job.log.Fields(logger.Fields{
+				"level":   userXP.Level,
+				"guildId": guildID,
+			}).Info("[updateLevelRoles] entity.GetUserRoleByLevel no role found")
+			continue
+		case err != nil:
```
Comment (trkhoi) [inline pkg/job/update_user_roles.go:96]: err != nil && err != gorm.ErrRecordNotFound

**Review by trkhoi — CHANGES_REQUESTED** [review]

**Review by trkhoi — APPROVED** [review]

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**